### PR TITLE
Match controller v2

### PR DESCRIPTION
**Pull request by eetar1**

### CODE REVIEW

File: web/v2/backend/src/main/java/com/tourneygen/web/Models/Services/MatchServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+  private void completionReport(Match match, Team victor, Team loser, MatchReportDTO reportDTO)
+      throws MatchConflictException {
+
+    if (match.getUpdatedBy().equals(reportDTO.getUpDatedBy())) {
+      throw new MatchConflictException(
+          "This teams has already posted to this match please wait for the other team to update");
+    }
+
+    // I only need to check one as the prior check handle the other
```
Comment (Ross-Alexandra) [inline web/v2/backend/src/main/java/com/tourneygen/web/Models/Services/MatchServiceImpl.java]: nit-pick, "I" don't do anything. Change to 3rd person (like an essay) plz

File: web/v2/backend/src/main/java/com/tourneygen/web/Models/Team.java [hunk 2]
```diff
@@ -173,4 +174,16 @@ public void merge(
                     () -> new EntityNotFoundException("League with id " + id + " was not found"));
     this.roster = teamUpdateDTO.getRoster() == null ? this.roster : teamUpdateDTO.getRoster();
   }
+
+  @Override
+  public boolean equals(Object o) {
```
Comment (Ross-Alexandra) [inline web/v2/backend/src/main/java/com/tourneygen/web/Models/Team.java]: overriding `equals` requires overriding`hashcode`.

I think I saw earlier that we're using maven. Look into the Commons Equals and Hashcode builders for doing this.

**Comment by Ross-Alexandra** [comment]
@eetar1 Pressed the wrong button
Changes required, not approved !!

File: web/v2/backend/src/main/java/com/tourneygen/web/Models/Team.java (hunk 2 above)
Comment (eetar1) [reply web/v2/backend/src/main/java/com/tourneygen/web/Models/Team.java]: Hashcode is done by the @Entity with the JPA functionality. When i change it it breaks the JPA functionality.

Comment (Ross-Alexandra) [reply web/v2/backend/src/main/java/com/tourneygen/web/Models/Team.java]: Fair enough

File: web/v2/backend/src/main/java/com/tourneygen/web/Models/Services/MatchServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+    if (!reportDTO.getVictorId().equals(match.getHomeTeam().getId())
+        && !reportDTO.getVictorId().equals(match.getAwayTeam().getId())) {
+      throw new IllegalArgumentException(
+          "The reported victor was not part of match " + match.getId());
+    }
+    if (!reportDTO.getLoserId().equals(match.getHomeTeam().getId())
+        && !reportDTO.getLoserId().equals(match.getAwayTeam().getId())) {
+      throw new IllegalArgumentException(
+          "The reported victor was not part of match " + match.getId());
```
Comment (Ross-Alexandra) [inline web/v2/backend/src/main/java/com/tourneygen/web/Models/Services/MatchServiceImpl.java]: The reported loser* ?

Comment (eetar1) [reply web/v2/backend/src/main/java/com/tourneygen/web/Models/Services/MatchServiceImpl.java]: Aw oops ik fix that

**Pull request merged**